syntax = "proto3";

package protos;

message CPUInfo {
    string arch = 1;
    int32 bits = 2;
    string brand = 3;
    int32 count = 4;
    string cpu_version = 5;
    int32 family = 6;
    int32 L2_cache_size = 7;
}

// Python Version
message PythonInfo {
    int32 major_version = 1;
    int32 minor_version = 2;
    int32 revision = 3;
}

message NetworkInfo {
    string ip_address = 1;
}

message SystemInfo {
    string hostname = 1;
    CPUInfo cpu_infos = 2;
    PythonInfo python_info = 3;
    NetworkInfo network_info = 4;
}